geo/src/algorithm/simplify.rs

```diff
@@ -293,13 +304,14 @@ mod test {
 
     #[test]
     fn recursion_test() {
-        let vec = [
+        let input = [
             coord! { x: 8.0, y: 100.0 },
             coord! { x: 9.0, y: 100.0 },
             coord! { x: 12.0, y: 100.0 },
         ];
-        let simplified = rdp::<_, _, 2>(vec.into_iter(), &1.0);
-        assert_eq!(simplified, vec);
+        let actual = rdp::<_, _, 2>(input.into_iter(), &1.0);
+        let expected = [coord! { x: 8.0, y: 100.0 }, coord! { x: 12.0, y: 100.0 }];
```

Comment:
This test was originally added in c61abfa1e758675cc6aa9925e3e6468cf104eaa4, and changed (I think incorrectly) in 9d81bf49ed31fdfeb76728563c8236879c2e2473 as part of #943

This reverts it to it's previous behavior.